src/core/or/circuitlist.c

```diff
@@ -2199,6 +2199,11 @@ circuit_mark_for_close_, (circuit_t *circ, int reason, int line,
   tor_assert(line);
   tor_assert(file);
 
+  /* Check whether the circuitpadding subsystem wants to block this close */
+  if (!circpad_circuit_should_be_marked_for_close(circ)) {
+    return;
+  }
```

Comment:
Should we be marking the circuit dirty and/or setting other flags that say "don't attach new streams here"? We don't want to change behavior and cause circuits that really should be not used to get used.

Reply:
Related: We should check the reason flag to see if this is a remote reason or some other condition that indicates the other side closed the circuit on us, or it is unusable. (This may require some grepping to verify ciircuit_mark_for_close() calls).